kedro-org/kedro-viz
# Support for Kedro 19

## Description

* This PR resolves the compatibility issues of Kedro Viz with Kedro 19
* [Kedro 19 Changes](https://github.com/kedro-org/kedro/blob/develop/RELEASE.md#release-0190)

## Development notes
* Update demo project to kedro version 19 (This is a partial update to run kedro viz tests against demo project. `kedro run` might not work on demo-project and needs more refactoring) 
* Remove _reformat_load_versions callback and import (This function is not available at `kedro.framework.cli.utils`)
* Update Dataset imports from `kedro.io.core`
* Update e2e tests to support older kedro versions and latest version (Requirements file [path change](https://github.com/kedro-org/kedro/blob/develop/RELEASE.md#other))
* Update e2e tests with new starter template `spaceflights-pandas` for latest version of kedro
* Fix lint issues related to Dataset imports
* Update datasets property access via DataCatalog `catalog._datasets`
* Remove try, except blocks in import statements for test files (eg., conftest) as we run pytests against latest kedro version
* Update DataCatalog initialization renaming data_sets to datasets and removing layers as mentioned [here](https://docs.kedro.org/en/stable/kedro.io.DataCatalog.html#kedro-io-datacatalog) 
* Update mock test api responses due to removal of layer and new id for TaskNode
* Update new way of passing `--params` to `kedro viz run` using `=` as mentioned [here](https://github.com/kedro-org/kedro/blob/develop/RELEASE.md#migration-guide-from-kedro-018-to-019)

## QA notes
* All tests should pass
* Create a new conda environment using `conda create --name <env_name> python=3.8` or clean old env requirements using `pip uninstall -y -r <(pip freeze)`
* Install requirements using `pip install -r package/test_requirements.txt -r demo-project/src/docker_requirements.txt`
* Run `make build`
* Run `pip install -e package`
* `cd demo-project`
* Run `kedro viz run`
* Kedro Viz should start successfully

## Checklist
